agreement/gossip/network.go

```diff
@@ -48,7 +48,8 @@ type networkImpl struct {
 	proposalCh chan agreement.Message
 	bundleCh   chan agreement.Message
 
-	net network.GossipNode
+	net           network.GossipNode
+	backgroundCtx context.Context
```

Comment:
What's the purpose of this `backgroundCtx` field?

Reply:
Our usage of the network library is limited; we never use the blocking version of it ( where the provided context would make a difference ).

So, instead of calling the context.Background() on every iteration, I cached that value.

I don't expect to see any notable gains here; this object is static anyway.

Reply:
It seems like something that would just be optimized out during compilation (see https://golang.org/src/context/context.go?s=7302:7327#L196). I think it makes more sense to just call it on every iteration so that we can get rid of this field.